inference-engine/src/mkldnn_plugin/nodes/mkldnn_gather_nd_node.h

```diff
@@ -42,19 +42,38 @@ class MKLDNNGatherNDNode : public MKLDNNNode {
     struct GatherNDExecutor {
         GatherNDExecutor(const GatherNDAttributes& attrs);
         ~GatherNDExecutor() = default;
-        void exec(const uint8_t* srcData, const int32_t* indices, uint8_t* dstData);
+        void exec(const MKLDNNMemoryPtr& srcMemPtr, MKLDNNMemoryPtr& dstMemPtr, const int32_t* indices);
```

Comment:
Just wondering why indices is not passed as memory pointer. Seems inconsistently.
Shouldn't it be `const MKLDNNMemoryPtr& dataMemPtr, const MKLDNNMemoryPtr& indicesMemPtr, MKLDNNMemoryPtr& dstMemPtr` or smt like that?

Reply:
Because `indices` are always `int32_t*`. It doesn't depend on implementation (blocks with memcpy or elementwise with template). I use memory pointer for src and dst data to avoid many casting from one data type to another. Because in implementation with memcpy we have `uint8_t*` pointers but in elementwise `T*`. We can use common `uint8_t*` pointers and then we should cast to `T*` data type in template method. But I think it's not good. 
Do you want suggest to use memory pointers for all parameters anyway?

Reply:
It is always (almost) better to write future proof code. Passing indices via MKLDNNMemoryPtr would allow to simplify future node extension for int64 precision. For sure this is not so important for now. Lets leave as is.